app/cosmo/src/main.rs

```diff
@@ -0,0 +1,457 @@
+        w.afr7().af12();
+        w
+    });
+    p.GPIOB.ospeedr.modify(|_, w| {
+        w.ospeedr7().very_high_speed();
+        w
+    });
+    p.GPIOB.moder.modify(|_, w| {
+        w.moder7().alternate();
+        w
+    });
+
+    // GPIOD
+    p.GPIOD.afrl.modify(|_, w| {
+        w.afr0().af12();
+        w.afr1().af12();
+        // pin 2 used for something else
+        w.afr3().af12();
+        w.afr4().af12();
+        w.afr5().af12();
+        w.afr6().af12();
+        w.afr7().af12();
+        w
+    });
+    p.GPIOD.afrh.modify(|_, w| {
+        w.afr8().af12();
+        w.afr9().af12();
+        w.afr10().af12();
+        w.afr11().af12();
+        w.afr12().af12();
+        w.afr13().af12();
+        w.afr14().af12();
+        w.afr15().af12();
+        w
+    });
+    p.GPIOD.ospeedr.modify(|_, w| {
+        w.ospeedr0().very_high_speed();
+        w.ospeedr1().very_high_speed();
+        // pin 2 used elsewhere
+        w.ospeedr3().very_high_speed();
+        w.ospeedr4().very_high_speed();
+        w.ospeedr5().very_high_speed();
+        w.ospeedr6().very_high_speed();
+        w.ospeedr7().very_high_speed();
+
+        w.ospeedr8().very_high_speed();
+        w.ospeedr9().very_high_speed();
+        w.ospeedr10().very_high_speed();
+        w.ospeedr11().very_high_speed();
+        w.ospeedr12().very_high_speed();
+        w.ospeedr13().very_high_speed();
+        w.ospeedr14().very_high_speed();
+        w.ospeedr15().very_high_speed();
+        w
+    });
+    p.GPIOD.moder.modify(|_, w| {
+        w.moder0().alternate();
+        w.moder1().alternate();
+        // pin 2 used elsewhere
+        w.moder3().alternate();
+        w.moder4().alternate();
+        w.moder5().alternate();
+        w.moder6().alternate();
+        w.moder7().alternate();
+
+        w.moder8().alternate();
+        w.moder9().alternate();
+        w.moder10().alternate();
+        w.moder11().alternate();
+        w.moder12().alternate();
+        w.moder13().alternate();
+        w.moder14().alternate();
+        w.moder15().alternate();
+        w
+    });
+
+    // GPIOE
+    p.GPIOE.afrl.modify(|_, w| {
+        w.afr0().af12();
+        w.afr1().af12();
+        // w.afr2().af12();
+        w.afr3().af12();
+        // w.afr4().af12();
+        // w.afr5().af12();
+        // w.afr6().af12();
+        w.afr7().af12();
+        w
+    });
+    p.GPIOE.afrh.modify(|_, w| {
+        w.afr8().af12();
+        w.afr9().af12();
+        w.afr10().af12();
+        w.afr11().af12();
+        w.afr12().af12();
+        w.afr13().af12();
+        w.afr14().af12();
+        w.afr15().af12();
+        w
+    });
+    p.GPIOE.ospeedr.modify(|_, w| {
+        w.ospeedr0().very_high_speed();
+        w.ospeedr1().very_high_speed();
+        // w.ospeedr2().very_high_speed();
+        w.ospeedr3().very_high_speed();
+        // w.ospeedr4().very_high_speed();
+        // w.ospeedr5().very_high_speed();
+        // w.ospeedr6().very_high_speed();
```

Comment:
remove commented out code?

Reply:
Good catch – we theoretically support a 24-bit memory address, but haven't gotten it working in practice.  I'll remove the commented-out code and add a note at the top.